### PR TITLE
[IMP] forum: display links of related forums in the sidebar

### PR DESCRIPTION
**Pull request by flch-odoo**
Display links of related forums in the sidebar

Task-3354382

### CODE REVIEW

File: addons/website_forum/controllers/website_forum.py [hunk 1]
```diff
@@ -38,7 +40,11 @@ def _prepare_user_values(self, **kwargs):
         if kwargs.get('forum'):
             values['forum'] = kwargs.get('forum')
         elif kwargs.get('forum_id'):
-            values['forum'] = request.env['forum.forum'].browse(int(kwargs.pop('forum_id')))
+            values['forum'] = Forum.browse(int(kwargs.pop('forum_id')))
+        forum = values.get('forum')
+        if isinstance(forum, Forum.__class__):
```
Comment (flch-odoo) [inline addons/website_forum/controllers/website_forum.py]:
```suggestion
        if isinstance(forum := values.get('forum'), Forum.__class__):
```

File: addons/website_forum/views/forum_forum_templates_layout.xml [hunk 2]
```diff
@@ -273,15 +273,13 @@
                     <i t-attf-class="fa fa-window-close fa-fw #{queue_type != 'close' and 'opacity-50'}"/> Closed
                 </a>
             </div>
-            <!-- [TODO] Fetch other forums' names and links if more than one forum -->
-            <div class="o_wforum_sidebar_section mb-3">
+            <div t-if="other_forums" class="o_wforum_sidebar_section mb-3">
```
Comment (flch-odoo) [inline addons/website_forum/views/forum_forum_templates_layout.xml]: I would use `forums_links` because we may not always be on a specific forum.

**Review by flch-odoo — COMMENTED** [review]
Thanks for taking up the task, @pyduf !
As discussed earlier, we will also at some point have the case where there is no "current" forum. See [WIP](https://github.com/odoo/odoo/pull/122926/files#diff-75faad674743bb2fe8e3daf88a623f02f4770fc86e0d1ffef0405a5869a3040dR101) where 
`forum` is `None`, but we could pass `forum="all"` to `_prepare_user_values`, so we do fetch forums links and include all accessible ones, right? 
If so, could you please already add this here, considering that we'll likely squash this small commit anyway?

File: addons/website_forum/controllers/website_forum.py (hunk 1 above)
Comment (pyduf) [reply addons/website_forum/controllers/website_forum.py]: Thanks for the tricks. But having added an elif based on forum (to match == "all"), I think it is better now to keep the variable before the if that is now a "switch case" based on the forum variable. What do you think ?

File: addons/website_forum/views/forum_forum_templates_layout.xml (hunk 2 above)
Comment (pyduf) [reply addons/website_forum/views/forum_forum_templates_layout.xml]: Indeed, now that it could contains all forum, other_forums is not a correct name. I have chosen "related_forums" as it is not links but forum instances (I need the whole instance to be able to call "slug" to build the link and of course the forum name for the link label). What do you think ?

**Comment by pyduf** [comment]
> Thanks for taking up the task, @pyduf ! As discussed earlier, we will also at some point have the case where there is no "current" forum. See [WIP](https://github.com/odoo/odoo/pull/122926/files#diff-75faad674743bb2fe8e3daf88a623f02f4770fc86e0d1ffef0405a5869a3040dR101) where `forum` is `None`, but we could pass `forum="all"` to `_prepare_user_values`, so we do fetch forums links and include all accessible ones, right? If so, could you please already add this here, considering that we'll likely squash this small commit anyway?

Thanks for the clarification and the link to the feature using the value forum == "all". For related forum, I only consider the following forum value: "all" and a specific forum. As mentioned in my first commit, the value can also be True. And you also have mentioned None here. For those values, in the current implementation, the related forums are not fetched. What do you think ? (We can fetch related forum also for value True and None but I don't think there are use cases for that).

**Comment by flch-odoo** [comment]
None is the value passed to the route, and in that case we would pass "all" here, that's what I meant.

What is `True` used for? (looks like it is overridden if there is a `forum` in `**post`, right?)

**Comment by pyduf** [comment]
> (looks like it is overridden if there is a `forum` in `**post`, right?)

The forum parameter of _prepare_user_values is set to True when called from _prepare_user_profile_values and there are 0 or more than one forum. In the forum (profile) template, it is used as a boolean to add forum tab in the user profile or not (questions, answers, activity, votes). But actually, it is always evaluated to True (either True or a recordset of one element). 

Anyway, in the case of user profile, the side bar is not displayed and the “related forums” are then not needed so it seems ok to let it that way (not loading related forum when “forum parameter” is True). 

To sum up, the “related forum” that are used to displays links to other forum in the sidebar are loaded when either a forum is passed as parameter (either as an id or a recordset) or when forum is “all”. In the first case, we load the forums different than the one passed, and in the latter case we load all the forums (of the website). That way, the forum links displayed in the side bar are links to forums different than the one selected (in case of “all”, no forum is selected so all forum links should be displayed).
